src/control/server/mgmt_svc.go

```diff
@@ -850,11 +858,82 @@ func (svc *mgmtSvc) KillRanks(ctx context.Context, req *mgmtpb.RanksReq) (*mgmtp
 			rrErr = nil
 		}
 		resp.Results = append(resp.Results,
-			NewRankResult(i.getSuperblock().Rank.Uint32(), "stop",
-				state, rrErr))
+			NewRankResult(i.getSuperblock().Rank.Uint32(), "stop", state, rrErr))
+	}
+
+	svc.log.Debugf("MgmtSvc.KillRanks dispatch, resp:%+v\n", *resp)
+
+	return resp, nil
+}
+
+func ping(i *IOServerInstance, rank uint32, timeout time.Duration) *mgmtpb.RanksResp_RankResult {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	resChan := make(chan *mgmtpb.RanksResp_RankResult, 1)
```

Comment:
This shouldn't be buffered, should it? As a rule of thumb most channels used for signaling between goroutines should be unbuffered in order to make it easier to reason about execution flow.